viewing_party/party.py

```diff
@@ -1,23 +1,175 @@
+    for fr_watched in user_data["friends"]:
+        for mov in fr_watched["watched"]:
+            set_mov_friends_titles.add(mov["title"])
+        
+    # create set of user movies titles
+    for movie in user_data["watched"]:
+        set_user_mov_titles.add(movie["title"])
+
+    # compare sets to determine only those that only user have seen
+    unique_titles = set_user_mov_titles.difference(set_mov_friends_titles)
+    
+    # add those unique movies to a list of movies based on title
+    for movie in user_data["watched"]:
+        if movie["title"] in unique_titles:
+            unique_watched.append(movie)
+
+    return unique_watched
+
+def get_friends_unique_watched(user_data):
+    unique_watched = []
+    set_user_mov_titles = set()
+    set_mov_friends_titles = set()
+
+    # create set of friends movies titles
+    for fr_watched in user_data["friends"]:
+        for mov in fr_watched["watched"]:
+            set_mov_friends_titles.add(mov["title"])
+
+    # create set of user movies titles
+    for movie in user_data["watched"]:
+        set_user_mov_titles.add(movie["title"])
+
+    # compare sets to determine only those that user have never seen
+    unique_titles = set_mov_friends_titles.difference(set_user_mov_titles)
+
+    # add those movies to the list based on titles
+    for fr_watched in user_data["friends"]:
+        for mov in fr_watched["watched"]:
+            if mov["title"] in unique_titles and mov not in unique_watched:
+                unique_watched.append(mov)
+            
+    return unique_watched
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+def get_available_recs(user_data):
+    friends_watched = get_friends_unique_watched(user_data)
+    recommended_movies = []
+    
+    # add to recommended_movies only those with available subscription
+    for movie in friends_watched:
+        if movie["host"] in user_data["subscriptions"]:
+            recommended_movies.append(movie)
+
+    return recommended_movies
+
 # -----------------------------------------
 # ------------- WAVE 5 --------------------
 # -----------------------------------------
 
+def get_new_rec_by_genre(user_data):
+    recommended_movies = get_friends_unique_watched(user_data)
+    recommended_by_genre = []
+    most_watched_genre = get_most_watched_genre(user_data)
+
+    # make a list of recommendations by genre
+    for movie in recommended_movies:
+        if movie["genre"] == most_watched_genre:
+            recommended_by_genre.append(movie)
+    
+    return recommended_by_genre
+
+def get_rec_from_favorites(user_data):
+    recommended_movies = get_unique_watched(user_data)
+    recommended_fav = []
+
+    # make a list of recommendations based on favorites
+    for movie in recommended_movies:
+        if movie in user_data["favorites"]:
+            recommended_fav.append(movie)
+    
+    return recommended_fav

```

Comment:
These last two functions look really good! Great use of the helper functions you have already written! These are very similar functions that have just enough differences to warrant two different functions!